snapcraft/remote/git.py

```diff
@@ -175,12 +181,20 @@ def push_url(
             "Pushing %r to remote %r with refspec %r.", ref, stripped_url, refspec
         )
 
+        # Create a list of tags to push
+        # From https://www.pygit2.org/references.html#pygit2.repository.References.iterator
+        # 2 -> GIT_REFERENCES_TAGS, fetches only tags
+        if push_tags:
+            tag_refs = [t.name for t in self._repo.references.iterator(2)]
```

Comment:
If you don't want the magic number there you can also reference it as:
```suggestion
        # Create a list of tags to push
        if push_tags:
            tag_refs = [t.name for t in self._repo.references.iterator(pygit2.GIT_REFERENCES_TAGS)]
```

Reply:
nice!